classes/db/Db.php

```diff
@@ -567,10 +567,10 @@ public function executeS($sql, $array = true, $use_cache = true)
 		if ($use_cache && $this->is_cache_enabled && $array)
 		{
 			$this->last_query_hash = Tools::encryptIV($sql);
-			if (($result = Cache::getInstance()->get($this->last_query_hash)) !== false)
+			if (Cache::getInstance()->exists($this->last_query_hash) !== false)
```

Comment:
@jocel1 Ok I understand it better now. I think that yout point is an additional argument in favor of using an Optional pattern approach to avoid this additional null-checking call using exists() function : 
* get() should return an Option object, 
* this Option object would be None if the value does not exist => replaces the exists() call.
* the Option object would also have a value() method to access the real value.

WDYT ?